src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp

```diff
@@ -73,11 +73,45 @@ class TBuildSingleAgentListInfoContext final : public AbstractHandler<std::share
         // Return elements should be one agent.
         if (response.size() == 1)
         {
-            const auto& agent = response.front();
-            data->m_agents.push_back({data->agentId().data(),
-                                      agent.at("name"),
-                                      Utils::leftTrim(agent.at("version"), "Wazuh "),
-                                      agent.at("ip")});
+            const auto clusterStatus = data->clusterStatus();
+            const auto clusterNodeName = data->clusterNodeName();
+            const auto agentId = data->agentId();
+
+            try
+            {
+                const auto& agent = response.front();
+                const auto& wdbAgentNodeName = agent.at("node_name").get_ref<const std::string&>();
+                const auto& wdbAgentConnectionStatus = agent.at("connection_status").get_ref<const std::string&>();
+                const auto& wdbAgentVersion = agent.at("version").get_ref<const std::string&>();
+                const auto& wdbAgentName = agent.at("name").get_ref<const std::string&>();
+                const auto& wdbAgentIp = agent.at("ip").get_ref<const std::string&>();
+
+                // If the agent is not part of a cluster or the agent is part of the cluster and the current agent
+                // node_name
+                // is the same as the server node name. Add the agent to the list of agents to be scanned.
```

Comment:
Quick simplification:
```suggestion
                // Add the agent to the scan list if one of the conditions is met:
                //   - It's not in a cluster 
                //   - It's in the cluster and matches the server node name.
```